packages/contact-center/user-state/src/helper.ts

```diff
@@ -15,8 +17,23 @@ export const useUserState = ({idleCodes, agentId, cc}) => {
       setElapsedTime(prevTime => prevTime + 1);
     }, 1000);
 
+    const handleStateChange = (data) => {
+      if (data && typeof data === 'object' && data.type === 'AgentStateChangeSuccess') {
+        const DEFAULT_CODE = '0'; // Default code when no aux code is present
+        setCurrentState({
+          id: data.auxCodeId?.trim() !== '' ? data.auxCodeId : DEFAULT_CODE
+        });
+        setElapsedTime(0);
+      }
+    };
+    
+    cc.on(AGENT_STATE_CHANGE, handleStateChange);
+
     // Cleanup the timer on component unmount
-    return () => clearInterval(timer);
+    return () => {
+      clearInterval(timer);
```

Comment:
nit. It's also beneficial to set `timer = null` to ensure there is no leak of timers. This is a recommendation I've seen before.